x/sync/syncable_db.go

```diff
@@ -3,11 +3,299 @@
 
 package sync
 
-import "github.com/ava-labs/avalanchego/x/merkledb"
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+
+	syncpb "github.com/ava-labs/avalanchego/proto/pb/sync"
+)
+
+var (
+	_ syncpb.SyncableDBServer = (*SyncableDBServer)(nil)
+	_ SyncableDB              = (*SyncableDBClient)(nil)
+)
 
 type SyncableDB interface {
 	merkledb.MerkleRootGetter
 	merkledb.ProofGetter
 	merkledb.ChangeProofer
 	merkledb.RangeProofer
 }
+
+func NewSyncableDBServer(db SyncableDB) *SyncableDBServer {
+	return &SyncableDBServer{db: db}
+}
+
+type SyncableDBServer struct {
+	syncpb.UnsafeSyncableDBServer
```

Comment:
is it better to avoid embedding this to ensure we implement all the methods?

Reply:
This is required to implement the interface and what we do in other places (e.g. `rpcdb.DatabaseServer`)